Item(by='mazeboard', descendants=None, kids=None, score=None, time=1603260073, title=None, item_type='comment', url=None, parent=24844915, text='handling errors could be done as follows:<p><pre><code>  (let ((err1 #{default:0})\n     (err2 #{default:#\\z})\n     (stack #{s:none,\n              err:none,\n       push:(fn x (set self.s (pair x self.s))),\n       pop:(fn ()\n      (if (null self.s)\n          then (error self.err)\n          else (let ((x (left self.s)))\n          in {\n            (set self.s (right self.s))\n            x\n          })))\n     })\n     (s1 #{stack, s:nil, err:err1})\n     (s2 #{stack, s:nil, err:err2}))\n        in  {\n        (handle-error\n         ((err1 (e) (print e.default))\n   (err2 (e) (print e.default)))\n         in {\n           (s1.push 5)\n    (print (s1.pop))\n    (print (s2.pop))})\n        })\n</code></pre>\nthe last pop from stack s2 will throw the error err2 and the handler will catch it and print the character #\\z.')